### ISSUE TITLE
Definition list audit doesn’t allow divs & violates the spec

### ISSUE THREAD

**Issue by toastal** [dequelabs/axe-core#4056]
### Product

axe-core

### Product Version

4.7.2 (Lighthouse)

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
A `<div>` wrapping definition list items is valid

#### Actual
Lighthouse flags me for invalid/inaccessible markup

#### How to Reproduce
```html
<dl>
	<div>
		<dt>1</dt>
		<dd>foo</dd>
		<dd>bar</dd>
	</div>
	<div>
		<dt>2</dt>
		<dd>qux</dd>
	</div>
</dl>
```

#### Additional context

> The [dl](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) element [represents](https://html.spec.whatwg.org/multipage/dom.html#represents) an association list consisting of zero or more name-value groups (a description list). A name-value group consists of one or more names ([dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) elements, possibly as children of a [div](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element) element child) followed by one or more values ([dd](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dd-element) elements, possibly as children of a [div](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element) element child), ignoring any nodes other than [dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) and [dd](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dd-element) element children, and [dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) and [dd](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dd-element) elements that are children of [div](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element) element children. Within a single [dl](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) element, there should not be more than one [dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) element for each name.
>
> &hellip;
>
> In order to annotate groups with [microdata](https://html.spec.whatwg.org/multipage/microdata.html#microdata) attributes, or other [global attributes](https://html.spec.whatwg.org/multipage/dom.html#global-attributes) that apply to whole groups, or just for styling purposes, each group in a [dl](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) element can be wrapped in a [div](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element) element. This does not change the semantics of the [dl](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) element.
> 
&ndash; https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element

Note that the spec specifically calls `<div>` elements for wrapping. There are even examples further down.

This div exception is missing from the docs as well: https://dequeuniversity.com/rules/axe/4.7/dlitem

There some sort of div check in here, but the code is hard to read
https://github.com/dequelabs/axe-core/blob/d0a49d33560b0c926c9c21836d5c4b29bdfb023c/lib/checks/lists/dlitem-evaluate.js#L10

**Comment by straker** [comment]
Interesting. Do you have an example page of this failing? Axe-core isn't reporting any accessibility issues with the code provided.

<img width="638" alt="Screenshot of Chrome DevTools showing the Elements panel with the provided code sample and the Console showing the results of an axe-core run with 0 incomplete and 0 violations." src="https://github.com/dequelabs/axe-core/assets/2433219/71b8441a-8426-4202-a918-1494ef25adc2">

**Comment by straker** [comment]
Thanks for that page. I believe what's going on is that the parent `dl` has a role of `contentinfo` so it is not acting as a definition list parent. If you remove the role it should work out.

```html
<dl role="contentinfo" class="Article-metadata">
```

**Comment by toastal** [comment]
Paraphrasing, is it because contentinfo turns an element into a footer? The definition list contains the metadata for the post so it *seems* like it should be the correct role. I could wrap it in a div (or footer after header), but adding extra elements for the landmark seems off.

**Comment by toastal** [comment]
Ah. It's not meant for that sort of metadata but like the literal page footer. The name is confusing to me as is the spec's definition.

**Comment by WilcoFiers** [comment]
Looks like this is resolved. Yeah I very much understand why "contentinfo" can be a confusing name. Thanks for reporting though! Always best to ask.

**Comment by toastal** [comment]
Yes, it is resolved.
